shell/platform/android/io/flutter/app/FlutterActivity.java

```diff
@@ -0,0 +1,143 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.app;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.Window;
+import android.view.WindowManager;
+import io.flutter.plugin.platform.PlatformPlugin;
+import io.flutter.view.FlutterMain;
+import io.flutter.view.FlutterView;
+import java.util.ArrayList;
+import org.chromium.base.TraceEvent;
+
+
+/**
+ * Base class for activities that use Sky.
+ */
+public class FlutterActivity extends Activity {
+    private FlutterView mView;
+
+    private String[] getArgsFromIntent(Intent intent) {
+        // Before adding more entries to this list, consider that arbitrary
+        // Android applications can generate intents with extra data and that
+        // there are many security-sensitive args in the binary.
+        ArrayList<String> args = new ArrayList<String>();
+        if (intent.getBooleanExtra("trace-startup", false)) {
+            args.add("--trace-startup");
+        }
+        if (intent.getBooleanExtra("start-paused", false)) {
+            args.add("--start-paused");
+        }
+        if (!args.isEmpty()) {
+            String[] argsArray = new String[args.size()];
+            return args.toArray(argsArray);
+        }
+        return null;
+    }
+
+    /**
+     * @see android.app.Activity#onCreate(android.os.Bundle)
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            Window window = getWindow();
+            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+            window.setStatusBarColor(0x40000000);
+            window.getDecorView().setSystemUiVisibility(PlatformPlugin.DEFAULT_SYSTEM_UI);
+        }
+
+        String[] args = getArgsFromIntent(getIntent());
+        FlutterMain.ensureInitializationComplete(getApplicationContext(), args);
+        mView = new FlutterView(this);
+        setContentView(mView);
+
+        onSkyReady();
+    }
+
+    /**
+     * @see android.app.Activity#onDestroy()
+     */
+    @Override
+    protected void onDestroy() {
+        if (mView != null) {
+            mView.destroy();
+        }
+        // Do we need to shut down Sky too?
```

Comment:
I guess I'll remove it, since we don't seem to have a way of shutting down in FlutterMain